### PR TITLE
Issue 6898: Update dependencies 

### LINKED ISSUE #6898: Upgrade dependencies with Security Vulnerabilities
**Describe the bug**
As per the latest scan, few of the dependencies are found to be vulnerable and needs to be updated.

**To Reproduce**
Run a security scan like BlackDuck or similar

**Location**
https://github.com/pravega/pravega/blob/master/gradle.properties

### PR DESCRIPTION
**Pull request by kotlasaicharanreddy**
Signed-off-by: SaiCharan <kotlasaicharan@yahoo.com>

**Change log description**  
Upgrades the following libraries to higher versions to fix CVEs found.

| Library | Current Version | Updated Version |
| --- | --- | --- |
hadoop | 3.3.2 | 3.3.4 |
com.fasterxml.jackson.core | 2.13.3 | 2.13.4
nettyVersion | 4.1.73.Final | 4.1.82.Final
nettyBoringSSLVersion | 2.0.48.Final | 2.0.54.Final
protobufProtocVersion | 3.19.4 | 3.21.7
AvroVersion |  | 1.11.1
KotlinVersion |  | 1.7.20
SnakeYamlVersion |  | 1.33
jdomVersion |  | 2.0.6.1

**Purpose of the change**  
Fixes #6898 

**What the code does**  
Bumps up libraries to higher versions.

**How to verify it**  
All tests must pass.

Following tests and compatibility have been done.
- System tests 1: 4820 is successful.
- Pravega-Spark-Connector Job: 71 is successful. 
- Tested flink-connectors locally and the build is successful.

### CODE REVIEW

**Review by krishan-Rai — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -123,13 +123,22 @@ allprojects {
...
             force "io.netty:netty-codec-socks:" + nettyVersion
             force "io.netty:netty-handler-proxy:" + nettyVersion
+            force "io.netty:netty-resolver-dns:" + nettyVersion
             force "io.netty:netty-transport-native-epoll:" + nettyVersion
             force "io.netty:netty-tcnative-boringssl-static:" + nettyBoringSSLVersion
             force "junit:junit:" + junitVersion
+            force "org.bouncycastle:bcprov-jdk15on:" + bouncyCastleVersion
+	    force "org.bouncycastle:bcpkix-jdk15on:" + bouncyCastleVersion
+	    force "org.apache.avro:avro:" +apacheAvroVersion
+	    force "org.jetbrains.kotlin:kotlin-stdlib-common:" +kotlinVersion
```
Comment (RaulGracia) [inline build.gradle:138]: Missing spaces after `+` in lines 137 to 141.

**Review by RaulGracia — CHANGES_REQUESTED** [review]
LGTM, just one minor comment. Also, please update the PR description and status.

**Comment by shshashwat** [comment]
TPC1 passed. Build no 4820

**Comment by shshashwat** [comment]
Spark connector is green too:- build 71

File: build.gradle (hunk 1 above)
Comment (kotlasaicharanreddy) [reply build.gradle]: Done. Updated description and status too.

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**